Framework/DataHandling/src/LoadNexusProcessed.cpp

```diff
@@ -577,33 +572,29 @@ void LoadNexusProcessed::correctForWorkspaceNameClash(std::string &wsName) {
 }
 
 /**
-* Check if the workspace name contains a common stem and load the workspace
-*names
+* Extract the workspace names from the file (if any are stored)
 *
 * @param root :: the root for the NeXus document
 * @param names :: vector to store the names to be loaded.
-* @return Whether there was a common stem.
 */
-bool LoadNexusProcessed::checkForCommonNameStem(
+void LoadNexusProcessed::extractWorkspaceNames(
     NXRoot &root, std::vector<std::string> &names) {
-  bool success(true);
   size_t nWorkspaceEntries = root.groups().size();
+  std::ostringstream os;
   for (size_t p = 1; p <= nWorkspaceEntries; ++p) {
-    std::ostringstream os;
+    os.str(std::string());
     os << p;
-
     names[p] = loadWorkspaceName(root, "mantid_workspace_" + os.str());
-
-    boost::smatch results;
-    const boost::regex exp(".*_\\d+$");
-
-    // check if the workspace name has an index on the end
-    if (!boost::regex_match(names[p], results, exp)) {
-      success = false;
-    }
   }
 
-  return success;
+  // Check that there are no duplicates in the workspace name
+  // This can cause severe problems
+  auto it = std::unique(names.begin(), names.end());
```

Comment:
The overall intention of the changes are to respect workspace name settings that are stored in a file. However, if you store two `mantid_workspace_XXX` entries with the same name (which you can easily do -- see the second test scenario), then selecting those names will cause issues with the ADS. Hence if we encounter a workspace name twice, then we ignore the stored workspace names and use the default means of workspace name selection. The default is triggered by not finding any stored names, ie a collection of empty strings -- see [here](https://github.com/mantidproject/mantid/blob/8f4873bfe2a60536c4ad993ada63f3347d7481f3/Framework/DataHandling/src/LoadNexusProcessed.cpp#L531:L541)

I will go ahead with the other changes.